src/components/MessagingWindow/MessagingWindow.jsx

```diff
@@ -0,0 +1,86 @@
+import React, { Component } from "react";
+import "./MessagingWindow.css";
+import Firebase from "firebase";
+import PropTypes from "prop-types";
+import Message from "./Message";
+
+class MessagingWindow extends Component {
+  constructor(props) {
+    super(props);
+
+    const { user, receiver } = this.props;
+    this.state = {
+      messages: [],
+      user,
+      receiver,
+      text: "" // temporary local variable used to handle text from the "send-message" text field
+    };
+  }
+
+  componentDidMount() {
+    const { user, receiver } = this.state;
+    this.tempRef = Firebase.database().ref(
+      `users/${user}/friends/${receiver}/`
+    ); // temporary reference to extract conversation reference
+    this.tempRef.on("value", snapshot1 => {
+      Firebase.database()
+        .ref(`/conversations/${snapshot1.val()}/messages/`)
+        .on("value", snapshot2 => {
+          this.setState({
+            messages: snapshot2.val() // gives complete list of messages in given conversation node
+          });
+          this.conversationRef = Firebase.database().ref(
+            `/conversations/${snapshot1.val()}/messages/`
+          );
+        });
+    });
+  }
+
+  handleChange = e => {
+    this.setState({ text: e.target.value });
+  };
+
+  handleClick = e => {
+    e.preventDefault();
+
+    const { text, user } = this.state;
+    const message = { text, user_id: user };
+    this.conversationRef.push(message);
+    this.setState({
+      text: ""
+    });
+  };
+
+  render() {
+    const { text: val, messages, user } = this.state;

```

Comment:
can you rename val to just text, i think it makes it a bit more readable. esp since val is kind of vague